slang-gfx.h

```diff
@@ -973,9 +1118,50 @@ class IResourceCommandEncoder : public ICommandEncoder
     virtual SLANG_NO_THROW void SLANG_MCALL
         uploadBufferData(IBufferResource* dst, size_t offset, size_t size, void* data) = 0;
 };
-#define SLANG_UUID_IResourceCommandEncoder                                             \
+
+enum class AccelerationStructureCopyMode
+{
+    Clone, Compact
+};
+
+struct AccelerationStructureQueryDesc
+{
+    QueryType queryType;
+
+    IQueryPool* queryPool;
+
+    int32_t firstQueryIndex;
+};
+
+class IRayTracingCommandEncoder : public ICommandEncoder
+{
+public:
+    virtual SLANG_NO_THROW void SLANG_MCALL buildAccelerationStructure(
+        const IAccelerationStructure::BuildDesc& desc,
+        int propertyQueryCount,
+        AccelerationStructureQueryDesc* queryDescs) = 0;
+    virtual SLANG_NO_THROW void SLANG_MCALL copyAccelerationStructure(
```

Comment:
Pedantically in COM interfaces the 'output' is typically the last parameter. It could be argued that this 'out' is really a modify so that's fine. I actually think the reverse order is better 

```
copyAccelerationStructure( mode, src, dst)
```

None of this is super important I'll grant.

Reply:
This is to stay consistent with the d3d api. The call maps 1:1 to d3d's BuildAccelerationStructure function so I'd just keep the interface the same.